pkg/operator/controller/nodeobservability/scc.go

```diff
@@ -66,39 +71,42 @@ func (r *NodeObservabilityReconciler) desiredSecurityContextConstraints(nodeObs
 	var priority int32 = 10
 
 	scc := &securityv1.SecurityContextConstraints{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: sccName,
-		},
+		ObjectMeta:               metav1.ObjectMeta{Name: sccName},
+		Priority:                 &priority,
 		AllowPrivilegedContainer: true,
-		AllowHostIPC:             false,
+		DefaultAddCapabilities:   nil,
+		RequiredDropCapabilities: []corev1.Capability{"MKNOD"},
+		AllowedCapabilities:      nil,
+		AllowHostDirVolumePlugin: true,
+		Volumes:                  []securityv1.FSType{securityv1.FSTypeHostPath, securityv1.FSTypeSecret, securityv1.FSTypeConfigMap},
 		AllowHostNetwork:         true,
-		AllowHostPID:             false,
 		AllowHostPorts:           false,
-		// This allows us to mount the hosts /var/run/crio/crio.sock into the container
-		AllowHostDirVolumePlugin: true,
-		AllowedCapabilities:      nil,
-		DefaultAddCapabilities:   nil,
-		FSGroup: securityv1.FSGroupStrategyOptions{
-			Type: securityv1.FSGroupStrategyRunAsAny,
-		},
-		Groups:                   []string{"system:cluster-admins", "system:nodes"},
-		Priority:                 &priority,
+		AllowHostPID:             false,
+		AllowHostIPC:             false,
+		SELinuxContext:           securityv1.SELinuxContextStrategyOptions{Type: securityv1.SELinuxStrategyMustRunAs},
+		RunAsUser:                securityv1.RunAsUserStrategyOptions{Type: securityv1.RunAsUserStrategyRunAsAny},
+		SupplementalGroups:       securityv1.SupplementalGroupsStrategyOptions{Type: securityv1.SupplementalGroupsStrategyRunAsAny},
+		FSGroup:                  securityv1.FSGroupStrategyOptions{Type: securityv1.FSGroupStrategyRunAsAny},
 		ReadOnlyRootFilesystem:   false,
-		RequiredDropCapabilities: []corev1.Capability{"MKNOD"},
-		RunAsUser: securityv1.RunAsUserStrategyOptions{
-			Type: securityv1.RunAsUserStrategyRunAsAny,
-		},
-		SELinuxContext: securityv1.SELinuxContextStrategyOptions{
-			Type: securityv1.SELinuxStrategyMustRunAs,
-		},
-		SupplementalGroups: securityv1.SupplementalGroupsStrategyOptions{
-			Type: securityv1.SupplementalGroupsStrategyRunAsAny,
-		},
-		Volumes: []securityv1.FSType{securityv1.FSTypeHostPath, securityv1.FSTypeSecret, securityv1.FSTypeConfigMap},
+		Groups:                   []string{"system:cluster-admins", "system:nodes"},
 	}
 	return scc
 }
 
+func (r *NodeObservabilityReconciler) updateSecurityContextConstraintes(ctx context.Context, current, desired *securityv1.SecurityContextConstraints) (bool, error) {
+	opts := cmpopts.IgnoreFields(securityv1.SecurityContextConstraints{}, "TypeMeta", "Users", "SeccompProfiles", "AllowedUnsafeSysctls", "ForbiddenSysctls")
+	if !cmp.Equal(current, desired, opts) {
+		// copy over current object metadata to updated
+		updated := desired.DeepCopy()
+		updated.ObjectMeta = current.ObjectMeta
```

Comment:
1) We should copy from the desired
2) `ObjectMeta` was the only thing not different in current and desired. We have to copy all the fields one by one. In the case the `cmp` package maybe not not even needed.